components/gitpod-protocol/src/gitpod-service.ts

```diff
@@ -285,6 +285,8 @@ export interface GitpodServer extends JsonRpcServer<GitpodClient>, AdminServer,
     getStripeSetupIntentClientSecret(): Promise<string>;
     findStripeSubscriptionId(attributionId: string): Promise<string | undefined>;
     getPriceInformation(attributionId: string): Promise<string | undefined>;
+    createStripeCustomer(params: GitpodServer.CreateStripeCustomerParams): Promise<void>;
+    /** @deprecated Use `GitpodServer.createStripeCustomer(GitpodServer.CreateStripeCustomerParams)` instead.  */
     createStripeCustomerIfNeeded(attributionId: string, currency: string): Promise<void>;
```

Comment:
@AlexTugarev Can we get away with a 3rd, optional parameter instead? If not, I feel we should just comment out the check (incl. a comment). :+1:

Reply:
No. JSON-RPC add an implicit `cancellationToken` argument. Adding a trailing optional parameter is discouraged, as omitting the argument would cause confusion, e.g. `lastOptionalButOmittedParam` would always be defined.

Reply:
Then let's go with uncommenting the check (to avoid this kind of noise on the API). WDYT?

Reply:
Which noice? That's effectively fixing API smell.